src/distro/distro.py

```diff
@@ -1241,14 +1244,32 @@ def _armbian_version(self) -> str:
         except FileNotFoundError:
             return ""
 
-    @staticmethod
-    def _parse_uname_content(lines: Sequence[str]) -> Dict[str, str]:
+    def _parse_uname_content(self, lines: Sequence[str]) -> Dict[str, str]:
         if not lines:
             return {}
         props = {}
-        match = re.search(r"^([^\s]+)\s+([\d\.]+)", lines[0].strip())
+        match = re.search(r"^([^\s]+)\s+([^\s]+)", lines[0].strip())
         if match:
-            name, version = match.groups()
+            name, release = match.groups()
+
+            # CloudLinux detection relies on uname release information
+            release_parts = release.split(".")
+            if (
+                # check penultimate release component contains an "el*" version
+                len(release_parts) > 1
+                and release_parts[-2].startswith("el")
+                and (
+                    # CloudLinux < 9 : "lve*" is set in kernel release
+                    any(rc.startswith("lve") for rc in release_parts)
+                    # CloudLinux >= 9 : check whether "kmodlve" is loaded
+                    or "kmodlve" in self._kernel_modules
+                )
+            ):
+                props["id"] = "cloudlinux"
+                props["name"] = "CloudLinux"
+                # strip "el" prefix and replace underscores by dots
+                props["release"] = release_parts[-2][2:].replace("_", ".")
+                return props
```

Comment:
Haven't looked at the full code base in awhile, but this is a lot of CloudLinux-specific hacks in one method. Is this something that we commonly do? Are there ways to better organize things where we can add additional hacks for other OSes without having to balloon this method?

Reply:
I'm torn, but I'll let you decide. I'm happy to help review CI stuff, but I'm much less familiar with the internals of distro and don't have access to these OSes to actually test things.

Reply:
Amended to move CloudLinux-related code in a dedicated method.